src/posts/2020-09-19-how-to-write-accessible-forms.md

```diff
@@ -29,69 +37,124 @@ tags:
 
-![A form with an input field highlighted in blue.](https://res.cloudinary.com/dgn6edv1k/image/upload/v1600204220/a11y_anz7bn.png)
+Form insertion [carets](https://developer.mozilla.org/en-US/docs/Glossary/caret#:~:text=A%20caret%20sometimes%20called%20a,is%20called%20the%20insertion%20point.) can be difficult to track. A simple solution is to highlight `<input>` on [focus](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus). This way, people do not have to guess where they are on the form. 
+
+![A form with an input field highlighted in blue.](../img/posts/2020-09-19-how-to-write-accessible-forms/form-field-on-focus.png)
 
 ```css
 input:focus{
-    outline: 2px solid royalblue
-    box-shadow: 1px 1px 8px 1px royalblue
+  outline: 2px solid royalblue;
+  box-shadow: 1px 1px 8px 1px royalblue;
 }
 ```
 
 ## Break long forms into smaller sections
-Filling long forms can be stressful. This is relatably true for almost everyone but perculiar to people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html).  They can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which is not ideal. A simple solution is to break them into smaller sections. These sections should include a progress bar.
 
-![A Multi page form with a progress bar at the top.](https://res.cloudinary.com/dgn6edv1k/image/upload/v1600197013/progress_wl7bgp.png)
+Filling long forms can be stressful. This is largely true for almost everyone but perculiarly for people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html). Such forms can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which isin't what anyone wants. A simple solution is to break them into smaller sections. 
+This helps make long forms less dauting and easier to understand as it is provided in bits.
+
+The following principles apply for multi-step forms:
+- Repeat overall instruction on every page.
+- The forms should be split into logical groups.
+- The forms should inform the person about the progress thay are making.
 
 
 ## Provide error messages if any
-If your site has forms, validation errors are inevitable. Always provide error messages when they occur. It is common practice to change the `<input>` color to red which is not ideal. Error messages helps people know exactly what they need to submit the form. 
 
-![A form with two fields indicating an error.](https://res.cloudinary.com/dgn6edv1k/image/upload/v1600197951/err_qiuycf.jpg)
+If your site has forms, validation errors are inevitable.
+When people encounter errors, clearly they need more help. Some people, like the visually impaired, may not even be aware that an error has occurred. It is common practice to change `<input>` color to red which is not ideal. This is because assistive devices will not catch the error, hence not informing the person of the error.

```

Comment:
```suggestion
When people encounter errors, they may need help to determine what's wrong and why. A common way forms indicate invalid fields is to change the `<input>` border color to red which is not ideal. The reason for this is because a change in color alone is not exposed to assistive technologies, so indicating errors in such ways will not inform the person of the error.

```